core/lib/zksync_core/src/api_server/web3/state.rs

```diff
@@ -638,6 +635,44 @@ impl Filters {
 
     /// Removes filter from the map.
     pub fn remove(&mut self, index: U256) -> bool {
-        self.state.remove(&index).is_some()
+        self.state.pop(&index).is_some()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn test_filters_functionality() {
+        use super::*;
+
+        let mut filters = Filters::new(2);
+
+        let filter1 = TypedFilter::Events(Default::default(), Default::default());
+        let filter2 = TypedFilter::Blocks(Default::default());
+        let filter3 = TypedFilter::PendingTransactions(Default::default());
```

Comment:
Bikeshedding: Using `Default::default()` is lazy; there's [a Clippy lint](https://rust-lang.github.io/rust-clippy/master/index.html#/default_trait_access) against it (we don't have it enabled, as many other lints).